app/models/info_request_batch.rb

```diff
@@ -117,4 +119,39 @@ def example_request
     end
     info_request
   end
+
+  # Do any of the requests in this batch have an embargo which is expiring
+  # soon?
+  #
+  # Returns boolean
+  def embargo_expiring?
+    info_requests.embargo_expiring.any?
+  end
+
+  # What phases are the requests in this batch in
+  #
+  # Returns unique array of symbols representing phases from InfoRequest::State
+  def request_phases
+    phases = info_requests.collect do |ir|
+      if ir.last_event_forming_initial_request_id.nil?
+        :awaiting_response
+      else
+        ir.state.phase
+      end
+    end
+    phases.uniq
+  end
+
+  # Summarise the phases requests are in into three groups:
+  # in progress, action needed, complete and provide a count of the number of
+  # requests in each group.
+  #
+  # Returns hash of string group names mapped to an integer
+  def request_phases_summary
+    {
+      _('In progress') => self.info_requests.in_progress.count,
+      _('Action needed') => self.info_requests.action_needed.count,
+      _('Complete') => self.info_requests.complete.count
```

Comment:
Ah, whoops. I looked through the list again and I wasn't so sure:
These could definitely be 'in progress'
`'gone_postal'`
`'internal_review'`

These too, although they're not quite so obvious:
`'error_message'`
`'requires_admin'`
`'attention_requested'`

But these two are kind of 'ending' statuses I think, you can't really do any more with a request if you've marked it as either of these?
`'vexatious'`
`'not_foi'`

Maybe a better solution would just be having an 'other' section to the progress bar instead, it could be a grey section at the end perhaps?